pkg/tenant/members_test.go

```diff
@@ -1 +1,83 @@
 package tenant_test
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/oklog/ulid/v2"
+	"github.com/rotationalio/ensign/pkg/tenant/api/v1"
+	"github.com/rotationalio/ensign/pkg/tenant/db"
+	"github.com/trisacrypto/directory/pkg/trtl/pb/v1"
+)
+
+func (suite *tenantTestSuite) TestMemberDetail() {
+	require := suite.Require()
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	member := &db.Member{
+		ID:   ulid.MustParse("01ARZ3NDEKTSV4RRFFQ69G5FAV"),
+		Name: "member-example",
+		Role: "Admin",
+	}
+	defer cancel()
+
+	// Connect to mock trtl database
+	trtl := db.GetMock()
+	defer trtl.Reset()
+
+	// Marshal the data with msgpack
+	data, err := member.MarshalValue()
+	require.NoError(err, "could not marshal the member")
+
+	// Unmarshal the data with msgpack
+	other := &db.Member{}
+	err = other.UnmarshalValue(data)
+	require.NoError(err, "could not unmarshal the member")
+
+	// Call the OnGet method and return test data.
+	trtl.OnGet = func(ctx context.Context, gr *pb.GetRequest) (*pb.GetReply, error) {
+		return &pb.GetReply{
+			Value: data,
+		}, nil
+	}
+
+	// Should return an error if the member does not exist.
+	_, err = suite.client.MemberDetail(ctx, "invalid")
+	suite.requireError(err, http.StatusBadRequest, "could not parse member id", "expected error when member does not exist")
+
+	// Create a member test fixture.
+	req := &api.Member{
+		ID:   "01ARZ3NDEKTSV4RRFFQ69G5FAV",
+		Name: "member-example",
+		Role: "Admin",
+	}
+	rep, err := suite.client.MemberDetail(ctx, req.ID)
+	require.NoError(err, "could not retrieve member")
+	require.Equal(req.ID, rep.ID, "expected member id to match")
+	require.Equal(req.Name, rep.Name, "expected member name to match")
+	require.Equal(req.Role, rep.Role, "expected member role to match")
+}
+
+func (suite *tenantTestSuite) TestMemberDelete() {
+	require := suite.Require()
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	memberID := "01ARZ3NDEKTSV4RRFFQ69G5FAV"
+
+	defer cancel()
+
+	// Connect to mock trtl database
+	trtl := db.GetMock()
+	defer trtl.Reset()
+
+	// Call the OnDelete method and return a DeleteReply.
+	trtl.OnDelete = func(ctx context.Context, dr *pb.DeleteRequest) (*pb.DeleteReply, error) {
+		return &pb.DeleteReply{}, nil
+	}
+
+	// Should return an error if the member does not exist.
+	err := suite.client.MemberDelete(ctx, "invalid")
+	suite.requireError(err, http.StatusBadRequest, "could not parse member id", "expected error when member does not exist")
```

Comment:
Can we add a test case for when the member doesn't exist? In other words, it's parseable but the trtl delete returns an error.

Reply:
I added a test below, but I'm not sure if this is exactly what you meant.

Reply:
I was thinking of something like this:

```go
trtl.OnDelete = func(ctx context.Context, dr *pb.DeleteRequest) (*pb.DeleteReply, error) {
    return nil, errors.New("key not found")
}

err = suite.client.MemberDelete(ctx, "missingID")
suite.RequireError(err, http.StatusNotFound, "could not delete member", "expected error when member ID is not found")
```

Reply:
Replacing `&pb.DeleteReply{}, nil` in the OnDelete function will result in the final error check returning an error. 

Also, when checking against `http.StatusNotFound` the following error message appears: 
`expected: "[404] could not delete member"`
`actual  : "[400] could not parse member id"`

Reply:
Gotcha, does it work if you move the new OnDelete and the new test case to the end of the method? Also my bad on the example, the ID needs to be a real ulid since the handler needs to parse it first.

Reply:
It works now. Thank you!